beacon-chain/rpc/beacon/assignments.go

```diff
@@ -101,42 +101,56 @@ func (bs *Server) ListValidatorAssignments(
 
 	shouldFetchFromArchive := requestedEpoch < bs.FinalizationFetcher.FinalizedCheckpt().Epoch
 
+	// initialize all committee related data.
+	committeeAssignments := map[uint64]*helpers.CommitteeAssignmentContainer{}
+	proposerIndexToSlot := map[uint64]uint64{}
+	archivedInfo := &pb.ArchivedCommitteeInfo{}
+	archivedBalances := []uint64{}
+
+	if shouldFetchFromArchive {
+		archivedInfo, err = bs.BeaconDB.ArchivedCommitteeInfo(ctx, requestedEpoch)
```

Comment:
I know this is not new code, but can you refactor everything within `shouldFetchFromArchive` into its function? 
```go
if shouldFetchFromArchive {
        bs.archivedCommitteeAssignment(ctx, requestedEpoch)
        ...
} else {
	committeeAssignments, proposerIndexToSlot, err = helpers.CommitteeAssignments(headState, requestedEpoch)
	if err != nil {
		return nil, status.Errorf(codes.Internal, "Could not compute committee assignments: %v", err)
	}
}
```